packages/legacy-client/src/client/objectSets/OsdkObjectSet.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ObjectTypesFrom, OntologyDefinition } from "@osdk/api";
+import type {
+  BaseObjectSetDefinition,
+  FilteredPropertiesTerminalOperations,
+  ObjectSetDefinition,
+} from "../../ontology-runtime";
+import type {
+  BaseObjectSet,
+  BaseObjectSetOperations,
+  ObjectSet,
+  ObjectSetOperations,
+} from "../interfaces";
+import type { SelectableProperties } from "../interfaces/utils/OmitProperties";
+import type { OsdkLegacyObjectFrom } from "../OsdkObject";
+import { createObjectSetAggregationStep } from "./createObjectSetAggregationStep";
+import { createObjectSetOrderByStep } from "./createObjectSetOrderByStep";
+import { createObjectSetSearchAround } from "./createObjectSetSearchAround";
+import { createObjectSetTerminalLoadStep } from "./createObjectSetTerminalLoadStep";
+
+export function createOsdkObjectSet<
+  O extends OntologyDefinition<any>,
+  K extends ObjectTypesFrom<O>,
+>(
+  objectSetDefinition: ObjectSetDefinition,
+  ontologyDefinition: O,
+): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+  const objectSet: ObjectSetOperations<OsdkLegacyObjectFrom<O, K>> = {
+    union(...otherObjectSets): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+      throw new Error("not implemented");
```

Comment:
We may need the definitions to be a property of the ObjectSet so we can introspect into it